shell/src/shell_automaton_manager.rs

```diff
@@ -147,20 +147,23 @@ impl ShellAutomatonManager {
             log.new(o!("service" => "quota")),
         );
 
-        let protocol_service =
-            ProtocolServiceDefault::new(mio_service.waker(), Arc::new(protocol_runner_api.clone()));
+        let prevalidator_service = PrevalidatorServiceDefault::new(
```

Comment:
@binier che line 74 above (github doesn't let me put the comment there). Would it be ok to wrap that enum value in `Box` as clippy suggests? it is huge